File: frontend/src/types/deliveryAddress.ts
export interface DeliveryAddress {
    _id: string;
    addressName: string;
    defaultAddress: boolean;
    nameBuyer: string;
    buyerId: string;
    phoneNumber?: string;
    note?: string;
    createdAt: string; // ISO Date string
    updatedAt: string; // ISO Date string
}

export interface CreateAddressPayload {
    addressName: string;
    nameBuyer: string;
    phoneNumber?: string;
    note?: string;
    defaultAddress?: boolean;
}

export type UpdateAddressPayload = Partial<CreateAddressPayload>;

export interface ApiResponse<T> {
    success: boolean;
    data: T;
    message?: string;
}

File: Mobileptit/app/src/main/java/com/ptitshop/app/data/model/DeliveryAddress.java
package com.ptitshop.app.data.model;

import com.google.gson.annotations.SerializedName;

public class DeliveryAddress {
    @SerializedName("_id")public String id;
    public String addressName;
    public Boolean defaultAddress;
    public String nameBuyer;
    public String buyerId;
    public String phoneNumber;
    public String note;
    public String createdAt;
    public String updatedAt;
}
